File: programmers/level2/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    count = Counter(tangerine)
    array = []
    for n, v in count.items():
        array.append((n, v))

    array.sort(key=lambda x: -x[1])

    for item in array:
        k -= item[1]
        answer += 1
        if k <= 0:
            break

    return answer

print(solution(6, [1, 3, 2, 5, 4, 5, 2, 3]))
print(solution(4, [1, 3, 2, 5, 4, 5, 2, 3]))
print(solution(2, [1, 1, 1, 1, 2, 2, 2, 3]))

File: programmers/level2/귤 고르기.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;

        Map<Integer, Integer> counter = new HashMap<>();
        for (int t: tangerine) {
            counter.put(t, counter.getOrDefault(t, 0) + 1);
        }

        List<Map.Entry<Integer, Integer>> sortedList = counter.entrySet()
            .stream()
            .sorted((e1, e2) -> e2.getValue().compareTo(e1.getValue()))
            .collect(Collectors.toList());

        for (Map.Entry<Integer, Integer> entry : sortedList) {
            k -= entry.getValue();
            answer += 1;
            if (k <= 0) {
                break;
            }
        }

        return answer;
    }
}